utilities/utilities.go

```diff
@@ -131,23 +129,40 @@ func SafeJoin(parent, subdir string) (string, error) {
 	return res, nil
 }
 
-// VerifyInput will verify an input against a signature
+// VerifyInput will verify an input against a signature using the public key.
 // A valid signature is indicated by returning a nil error.
-func VerifyInput(input string, signature string) error {
+func VerifyInput(input string, signature string, pubKey *rsa.PublicKey) error {
 	sign, _ := hex.DecodeString(signature)
-	block, _ := pem.Decode([]byte(globals.SignatureKey))
+	h := sha256.New()
+	h.Write([]byte(input))
+	d := h.Sum(nil)
+	return rsa.VerifyPKCS1v15(pubKey, crypto.SHA256, d, sign)
+}
+
+// ParseRsaPublicKey parses a public key in PEM format and returns the rsa.PublicKey object.
+// Returns an error if the key is invalid.
+func ParseRsaPublicKey(key []byte) (*rsa.PublicKey, error) {
+	block, _ := pem.Decode(key)
 	if block == nil {
-		return errors.New("invalid key")
+		return nil, errors.New("invalid key")
+	}
+
+	parsedKey, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return nil, err
 	}
-	key, err := x509.ParsePKIXPublicKey(block.Bytes)
+
+	return parsedKey.(*rsa.PublicKey), nil
```

Comment:
I would check if the key is an rsa key
```suggestion
 publicKey, ok := value.(*rsa.PublicKey)
    if !ok {
        return nil,fmt.Errorf("not an rsa key")
    }
return publicKey, nil
```